Add a websocket proxy to support JSMPEG on the Frigate Lovelace card

dermotduffy opened the pull request:
* Add a general websocket proxy that we can use to proxy between HA <-> Frigate.
 * Instantiate the (authenticated) proxy for JSMPEG proxying.
 * Required to support: https://github.com/dermotduffy/frigate-hass-card/issues/18

dermotduffy commented:
@blakeblackshear when you have a moment, would appreciate a second pair of eyes on this.

blakeblackshear commented:
I don't have any concerns. I didn't test it because it will be easier once the lovelace card is updated. Did you reference how the ingress component proxies websockets?

dermotduffy commented:
Dang. No, I didn't, I implemented it from first principles (which took much longer, most likely). I've now compared against the ingress WS proxy -- their code is strikingly similar, I updated this PR with some bells, whistles and cleanups that the ingress implementation had that mine didn't (protocol selection & passing the query string through being the main two) -- thanks for the idea to compare.

I have an updated version of the lovelace card working against this reasonably well, so I'll go with this for now. Thanks for the review as always!
